### PR TITLE
[CI/Build] Drop Python 3.8 support

### PR DESCRIPTION
**Pull request by russellb**
706e3609 [CI/Build] Drop Python 3.8 support
bbdf33a7 [Core] Make use of str.removeprefix()

commit 706e3609221eccc8ea03baf5b90e6e5a7ea4b758
Author: Russell Bryant <rbryant@redhat.com>
Date:   Tue Nov 5 09:23:59 2024 -0500

    [CI/Build] Drop Python 3.8 support
    
    Pytorch 3.5 dropped support for Python 3.8, so it's time we do the
    same.
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>

commit bbdf33a7a425640f184618c4bba04d9a3c4fffc5
Author: Russell Bryant <rbryant@redhat.com>
Date:   Tue Nov 5 09:29:37 2024 -0500

    [Core] Make use of str.removeprefix()
    
    Previously we had a custom implementation of this to support Python
    3.8 since this method was added in Python 3.9. Now that we have
    dropped 3.8, we can use the built-in method for this.
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>

### CODE REVIEW

**Comment by russellb** [comment]
> There is an existing PR #8464 but the author seems too busy to handle this.

gotcha - I didn't see that one. Unless the author picks it back up, I'm happy to take it over.

**Comment by DarkLight1337** [comment]
Quick heads-up that the author has picked up the other PR.

File: docs/source/getting_started/installation.rst [hunk 1]
```diff
@@ -66,7 +66,7 @@ If you want to access the wheels for previous commits, you can specify the commi
     $ export VLLM_COMMIT=33f460b17a54acb3b6cc0b03f4a17876cff5eafd # use full commit hash from the main branch
     $ pip install https://vllm-wheels.s3.us-west-2.amazonaws.com/${VLLM_COMMIT}/vllm-1.0.0.dev-cp38-abi3-manylinux1_x86_64.whl
 
-Note that the wheels are built with Python 3.8 ABI (see `PEP 425 <https://peps.python.org/pep-0425/>`_ for more details about ABI), so **they are compatible with Python 3.8 and later**. The version string in the wheel file name (``1.0.0.dev``) is just a placeholder to have a unified URL for the wheels. The actual versions of wheels are contained in the wheel metadata.
+Note that the wheels are built with Python 3.9 ABI (see `PEP 425 <https://peps.python.org/pep-0425/>`_ for more details about ABI), so **they are compatible with Python 3.9 and later**. The version string in the wheel file name (``1.0.0.dev``) is just a placeholder to have a unified URL for the wheels. The actual versions of wheels are contained in the wheel metadata.
```
Comment (DarkLight1337) [inline docs/source/getting_started/installation.rst:69]: Oh, I think I missed this paragraph in the other PR.

**Comment by russellb** [comment]
> I have added some things that were missed in other PR to this PR. See if this looks good to you @russellb

lgtm!

**Review by DarkLight1337 — APPROVED** [review]

File: docs/source/getting_started/installation.rst (hunk 1 above)
Comment (youkaichao) [reply docs/source/getting_started/installation.rst:69]: oh this should actually be reserved.

**Pull request merged**